Issue 2694 update schema design docs (#3215)
IBM/FHIR: fhir-persistence-schema/docs/DB2MultiTenancy.md

**Table: Resources**

> Each resource is contains a RESOURCE and LOGICAL RESOURCE table. `xx_LOGICAL_RESOURCES` has one record per logical resource.  `xx_RESOURCES` has one record for each version of each logical resource. 

Reviewer: ```suggestion
Each resource type has its own RESOURCE and LOGICAL RESOURCE tables. `xx_LOGICAL_RESOURCES` has one record per logical resource.  `xx_RESOURCES` has one record for each version of each logical resource. 
```

Author: ok
